src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -336,12 +334,18 @@ public void saveAs(File file) throws Exception {
             return;
         }
 
-        //closing AutosaveManager and BackupManager for original library 
-        context.setDatabaseFile(oldFile.toFile());
-        AutosaveManager.shutdown(context);
-        BackupManager.shutdown(context);
+        try {
+            final Path oldFile = context.getDatabasePath().get();
+            context.setDatabaseFile(oldFile.toFile());
+            //closing AutosaveManager and BackupManager for original library
+            AutosaveManager.shutdown(context);
+            BackupManager.shutdown(context);
+        } catch (NoSuchElementException e) {
+            LOGGER.info("Old file not found, just creating a new file", e);
```

Comment:
`context.getDatabasePath().get()` returns an `Optional` which you can check for being present/absent using `isPresent()`. Then you won't need a try-catch block.

Check out this page to learn the most important features of `Optional`s https://examples.javacodegeeks.com/core-java/util/optional/java-8-optional-example/